Cargo.toml

```diff
@@ -55,6 +56,7 @@ default = [
 
 [dependencies]
 num-traits = { version = "0.2.15", optional = true }
+time = { version = "0.3.5", features = ["local-offset"] }
```

Comment:
I know, another dependency. `time` is used in other examples, and the `local-offset` works nicely for generating "ticks" - the time `millisecond()` method conveniently returns a number between 0 and 1,000. `chrono` appears to be the current time lib, so if there's a similar way to implement the functionality with `chrono`, I can switch.

Reply:
I guess take a look at https://docs.rs/chrono/latest/chrono/struct.DateTime.html  and let me know if this allows you to do what you need to do with this addition. Otherwise If it does not Please add time as optional behind a feature.